test/e2e_node/dockershim_checkpoint_test.go

```diff
@@ -0,0 +1,210 @@
+			runPodCheckpointTest(f, func() {
+				checkpoints := findCheckpoints(f.Namespace.Name)
+				if len(checkpoints) == 0 {
+					framework.Failf("No checkpoint for the pod was found")
+				}
+				By("Removing checkpoint of test pod")
+				for _, filename := range checkpoints {
+					filename = cleanUpOutput(filename)
+					if len(filename) == 0 {
+						continue
+					}
+					framework.Logf("Removing checkpiont %q", filename)
+					_, err := exec.Command("sudo", "rm", filename).CombinedOutput()
+					framework.ExpectNoError(err, "Failed to remove checkpoint file %q: %v", string(filename), err)
+				}
+			})
+		})
+	})
+
+	Context("When pod containers are missing", func() {
+		It("should complete pod sandbox clean up based on the information in sandbox checkpoint", func() {
+			runPodCheckpointTest(f, func() {
+				By("Gathering pod container ids")
+				stdout, err := exec.Command("sudo", "docker", "ps").CombinedOutput()
+				framework.ExpectNoError(err, "Failed to run docker ps: %v", err)
+
+				ids := []string{}
+				lines := strings.Split(string(stdout), "\n")
+				for _, line := range lines[1:] {
+					if strings.Contains(line, f.Namespace.Name) {
+						ids = append(ids, strings.Fields(line)[0])
+					}
+				}
+
+				By("Stop and remove pod containers")
+				dockerStopCmd := append([]string{"docker", "stop"}, ids...)
+				_, err = exec.Command("sudo", dockerStopCmd...).CombinedOutput()
+				framework.ExpectNoError(err, "Failed to run command %v: %v", dockerStopCmd, err)
+				dockerRmCmd := append([]string{"docker", "rm"}, ids...)
+				_, err = exec.Command("sudo", dockerRmCmd...).CombinedOutput()
+				framework.ExpectNoError(err, "Failed to run command %v: %v", dockerRmCmd, err)
+			})
+		})
+	})
+
+	Context("When checkpoint file is corrupted", func() {
+		It("should complete pod sandbox clean up", func() {
+			runPodCheckpointTest(f, func() {
+				By("Corrupt checkpoint file")
+				checkpoints := findCheckpoints(f.Namespace.Name)
+				if len(checkpoints) == 0 {
+					framework.Failf("No checkpoint for the pod was found")
+				}
+				for _, file := range checkpoints {
+					f, err := os.OpenFile(file, os.O_WRONLY|os.O_APPEND, 0644)
+					framework.ExpectNoError(err, "Failed to open file %q", file)
+					_, err = f.WriteString("blabblab")
+					framework.ExpectNoError(err, "Failed to write to file %q", file)
+					f.Sync()
+					f.Close()
+				}
+			})
+		})
+	})
+})
+
+func runPodCheckpointTest(f *framework.Framework, twist func()) {
+	By(fmt.Sprintf("Creating test pod: %s", testPod))
+	f.PodClient().CreateSync(&v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{Name: testPod},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Image: "gcr.io/google_containers/busybox:1.24",
+					Name:  "checkpoint-test-container",
+					Command: []string{
+						"sh",
+						"-c", // Make 100 billion small files (more than we have inodes)
+						"sleep 36000",
+					},
+				},
+			},
+		},
+	})
+
+	By("Performing disruptive operations")
```

Comment:
Shouldn't you you perform the sanity check in line 146 before performing disruptive operations?